### ISSUE TITLE
Suggestion: Make dependencies provided

### ISSUE THREAD

**Issue by sbglasius** [virtualdogbert/GroovyConfigWriter#2]
Right now the plugin demands `compile` dependencies on the following:
```
 +--- com.virtualdogbert:GroovyConfigWriter:0.1
 |    +--- org.codehaus.groovy:groovy-all:2.4.11 -> 2.4.10
 |    +--- org.slf4j:slf4j-api:1.6.1 -> 1.7.25
 |    \--- ch.qos.logback:logback-classic:0.9.28 -> 1.1.11 (*)
``` 
Perhaps it would be a better idea to make them `provided` that way I don't think they will influence the projects where they are used. I'm not sure if it's an issue at all, but to be on the safe side.

**Comment by virtualdogbert** [comment]
So this is not a plugin, just a stand alone groovy library. I tried provided, that's not available, but I can substitute compileOnly, which should be equivalent. So if I made this change I would have to update the documentation, to tell users they have to provide these dependencies in there project, otherwise they would get  NoClassDefFoundErrors. A user could also use excludes, on the dependencies, if they cause a conflict. On the other hand these dependencies would probably be common in any project they are used.

So basically I've arguing with myself here, and I'm not really convinced that one way is necessarily  "better", than another.  What do you think?

**Comment by virtualdogbert** [comment]
Change dependencies to compile only in 1.0
